### PR TITLE
Add --debug-skip-weight-update and --debug-disable-optimizer

### PR DESCRIPTION
**Pull request by yueming-yuan**
## Summary
Two narrow debug knobs surfaced during DeepSeek-V4 Pro 1.6T RL bring-up. Both default off; behavior with neither flag set is byte-identical to before.

### `--debug-skip-weight-update`
Preserves the train/rollout offload-onload schedule and the rollout-engine recovery / process-group reload paths, but skips the actual actor-to-rollout weight transfer. Used to isolate the weight-sync code path (CUDA IPC / NCCL broadcast bucketing) from the rest of the RL inner loop.

### `--debug-disable-optimizer`
Skips Megatron optimizer + LR scheduler initialization and skips `optimizer.prepare_grads`/`optimizer.step`/`scheduler.step` in `train_one_step`. Forces `no_load_optim=True` and `no_save_optim=True` so checkpoint I/O stays consistent with the missing optimizer state. Used to validate the full RL loop (rollout, log_probs, actor forward/backward, weight update, offload/onload) without paying for optimizer state allocation, and to A/B-test whether a failure is optimizer-side or elsewhere.

`setup_model_and_optimizer` / `train_one_step` / `train` / `save` / `initialize_model_and_optimizer` now accept `MegatronOptimizer | None` and `OptimizerParamScheduler | None` to thread the no-optimizer case through.

## Test plan
- [ ] CI passes with both flags off (default → no behavior change)
- [ ] Smoke run with `--debug-skip-weight-update`: train loop completes the rollout / log_probs / offload-onload cycle without sending weight tensors
- [ ] Smoke run with `--debug-disable-optimizer`: train loop completes rollout + actor forward/backward; logs `Skipping optimizer.prepare_grads(), optimizer.step(), and LR scheduler step` and never errors on missing optimizer state

### CODE REVIEW

File: miles/backends/megatron_utils/model.py [hunk 1]
```diff
@@ -345,11 +355,13 @@ def train_one_step(
...
     args = get_args()
     dumper_phase_util = DumperMegatronUtil(args, model, DumperPhase.FWD_BWD)
+    disable_optimizer = getattr(args, "debug_disable_optimizer", False) or optimizer is None
 
     # Set grad to zero.
     for model_chunk in model:
         model_chunk.zero_grad_buffer()
-    optimizer.zero_grad()
+    if not disable_optimizer:
+        optimizer.zero_grad()
```
[the remark this answers is not in the document]
Comment (yueming-yuan) [reply miles/backends/megatron_utils/model.py:364]: this should not be a issue

**Comment by Zhichenzzz** [comment]
quick reminder: there is one CI failed (https://github.com/radixark/miles/actions/runs/25204569672/job/73902444553?pr=1053#step:9:4035)

**Review by Zhichenzzz — APPROVED** [review]

**Comment by Zhichenzzz** [comment]
LGTM! Thanks!
